gammapy/cube/fit.py

```diff
@@ -267,6 +268,63 @@ def npred(self):
 
         return npred_total
 
+    @classmethod
+    def create(cls, geom, geom_irf=None, migra_axis=None, rad_axis=None):
+        """Creates a MapDataset object with zero filled maps
+
+        Parameters
+        ----------
+        geom: `~gammapy.maps.WcsGeom`
+            Reference target geometry in reco energy, used for counts and background maps
+        geom_irf: `~gammapy.maps.WcsGeom`
+            Reference image geometry in true energy, used for IRF maps.
+        migra_axis: `~gammapy.maps.MapAxis`
+            Migration axis for the energy dispersion map
+        rad_axis: `~gammapy.maps.MapAxis`
+            Rad axis for the psf map
+        """
+
+        geom_irf = geom_irf or geom
+
+        counts = Map.from_geom(geom, unit="")
+
+        background = Map.from_geom(geom, unit="")
+        background_model = BackgroundModel(background)
+
+        energy_axis = geom_irf.get_axis_by_name("ENERGY")
+
+        exposure_geom = geom.to_image().to_cube([energy_axis])
```

Comment:
Here you copy the spatial part of the counts map, so if `geom_irf` is defined on a larger spatial footprint the information is lost here. Maybe just upsample the exposure?

Reply:
This is the final exposure map, which I believe should be on the same spatial geom as the counts map, but with a true energy axis.

Reply:
To handle spill-over of the PSF correctly at the edges of the predicted counts map, the exposure map also needs to be larger (or have a margin of the width of the PSF). So I think it would make more sense to create it from the `geom_irf` and upsample it to the binning of the counts map...

Reply:
Sorry @AtreyeeS! Please ignore my previous comment, I realised our model evaluation list not yet ready to handle the case of the larger exposure map completely. So introducing it here, would require additional changes in the model evaluation. Let's do it a later PR consistently.